### PR TITLE
Set association

### PR DESCRIPTION
**Pull request by joskalenda**
## OOP school library: set up associations

- [ ] Learning objectives

- Run a program using the command line.

- Set up associations between classes and objects.

### Features

- [ ] Create a class Classroom with the following:

1. @label instance variable, should be initialized in the constructor.
2. Setter and getter for @label (remember about attr_accessor).
3. Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
4. Create the has-many side (a classroom has many students).
5. Create the belongs-to side (a student belongs to a classroom).
6. Make sure that when adding a student to a classroom it also sets the classroom for the student.
7. Make sure that when setting the classroom for a student it also adds it to the classrooms' students.

- [ ] Create a class Book with the following:

1. @title and @author instance variables should be initialized in the constructor.
2. Setters and getters for instance variables (remember about attr_accessor).

- [ ] Create a class Rental with the following:

1. @date instance variable should be initialized in the constructor.
2. Setter and getter for @date (remember about attr_accessor).
3. Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
4. Create the has-many side of Book and Rental (a book has many rentals).
5. Create the belongs-to side of Rental and Book (a rental belongs to a book).
6. Create the has-many side of Person and Rental (a person has many rentals).
7. Create the belongs-to side of Rental and Person (a rental belongs to a person).
8. Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: rental.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Rental
+  attr_accessor :date
```
Comment (medaminedev66) [inline rental.rb]: - [ ] I would recommend adding Getters and Setters for @book and @person instance variables.

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @joskalenda  😃, :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Your code organization was really good 👍

## Required Changes ♻️
- [ ] Following the project requirements, A person should have many rentals. Kindly create the `has-many` side of Person and Rental.
- [ ] Kindly add a method to the Student class that serves as the 'belongs-to' side (a student belongs to a classroom). Check out this article with examples: https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships to learn more about both ways relationship.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: rental.rb (hunk 1 above)
Comment (joskalenda) [reply rental.rb]: Thanks, @medaminedev66 , I do the changes in a while

**Review by enaburekhan — APPROVED** [review]
Hi, joskalenda, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Code review by **Eric Enaburekhan**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by joskalenda** [comment]
Finally 😄 , Thanks, brother @enaburekhan

**Pull request merged**